frontend/src/components/doctor/DetailDoctorPage.tsx

```diff
@@ -32,22 +26,9 @@ export default function DetailDoctorPage() {
   // get request
   // https://beta.reactjs.org/learn/synchronizing-with-effects#fetching-data
   // you need to add last name to the url
-  useEffect(() => {
-    let ignore = false;
-    const getIndividualDoctor = async () => {
-      const request = await fetch(`/api/doctor/${userId}`);
-      const result = (await request.json()) as Doctor;
-
-      console.log(result);
-      if (!ignore) {
-        setIndividualDoctor(result);
-      }
-    };
 
-    getIndividualDoctor();
-    return () => {
-      ignore = true;
-    };
+  useEffect(() => {
+    (async () => setIndividualDoctor(await getIndividualDoctor(userId)))();
```

Comment:
You still need the ignore logic here or you'll have the same issue where if you navigate away from the page before the network request finishes, you'll get an error because React will try to set state in a component that isn't mounted anymore.